java/core/src/main/java/com/microsoft/bond/ListBondType.java

```diff
@@ -102,6 +103,11 @@ public final boolean isGenericType() {
         return new LinkedList<TElement>();
     }
 
+    private static <TElement> List<TElement> newDefaultValue(int initialCapacity) {
+        // custom initial capacity to match element count when deserializing
+        return new ArrayList<TElement>(initialCapacity);
+    }
```

Comment:
Why are you using ArrayList for list? We've decided to use ArrayList for vectors and LinkedList for lists.